xbmc/utils/UpdateHandler.cpp

```diff
@@ -0,0 +1,150 @@
+/*
+ *      Copyright (C) 2015 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+ 
+#include "UpdateHandler.h"
+#include "IUpdater.h"
+#include "settings/lib/Setting.h"
+#include "settings/Settings.h"
+#include "ServiceBroker.h"
+
+#if defined(TARGET_DARWIN_OSX)
+#include "platform/darwin/osx/UpdaterOsx.h"
+#elif defined(TARGET_WINDOWS) && !defined(TARGET_WIN10)
```

Comment:
afaik `TARGET_WINDOWS` isn't set for uwp builds, but @afedchin can you confirm
same is done a few lines below
also I wouldn't care about uwp, because it's not in master yet